src/main/java/ui/MenuControl.java

```diff
@@ -317,6 +325,59 @@ private MenuItem createRefreshMenuItem() {
         return new MenuItem[] { newIssueMenuItem, newLabelMenuItem, newMilestoneMenuItem };
     }
 
+    private MenuItem[] createReposMenu() {
+        Menu remove = new Menu("Remove");
+
+        ui.registerEvent((OpenReposChangedEventHandler) e -> {
+            Platform.runLater(() -> updateRepoRemoveList(remove));
+        });
+
+        return new MenuItem[] { remove };
+    }
+
+    private void updateRepoRemoveList(Menu remove) {
+        remove.getItems().clear();
+
+        HashSet<String> currentlyUsedRepo = getCurrentlyUsedRepo();
+
+        for (String repoId : ui.logic.getStoredRepos()) {
+            if (!currentlyUsedRepo.contains(repoId)) {
+                MenuItem removeItem = new MenuItem(repoId);
+                removeItem.setOnAction(e1 -> onRepoRemove(repoId));
+                remove.getItems().add(removeItem);
+            }
+        }
+    }
+
+    private HashSet<String> getCurrentlyUsedRepo() {
+        HashSet<String> currentlyUsedRepos = new HashSet<String>();
```

Comment:
No, it's not deleted when it isn't referenced. If a repo is ever being used in the current session, it's marked as open throughout the whole session, even when nothing is using it anymore (say panel is closed).

As such, if a user clears a `repo` filter, the repo status is still open. There is no way of telling what `repo` filter was just cleared either. As such, I need to traverse panels to check what repos are currently in use.